haystack/modeling/training/base.py

```diff
@@ -630,7 +627,7 @@ class DistillationTrainer(Trainer):
     """
     def __init__(
         self,
-        model: "FARMReader",
+        model: "AdaptiveModel",
```

Comment:
Is the code ready to use other models than FARMReader in its current form?

Reply:
It can basically train any AdaptiveModel with a QA `prediction_head`. I changed this line because I realised that _training_procedure only passes the AdaptiveModel. This behavior is exactly the same for the normal `Trainer` class.